### ISSUE TITLE
Error in importing optimum neuron: cannot import name 'GenerationMode' from 'transformers.generation.utils'

### ISSUE THREAD

**Issue by nd7141** [huggingface/optimum-neuron#210]
<img width="256" alt="image" src="https://github.com/huggingface/optimum-neuron/assets/4607709/caeaa10b-66b7-4113-9fb8-c7a540e4b1bb">

```
from transformers import AutoTokenizer
from optimum.neuron import NeuronModelForCausalLM
```

results in 

```
RuntimeError: Failed to import optimum.neuron.modeling because of the following error (look up to see its traceback):
cannot import name 'GenerationMode' from 'transformers.generation.utils' (/opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/transformers/generation/utils.py)
```

**Comment by nd7141** [comment]
I just followed this snippet: https://huggingface.co/docs/optimum-neuron/guides/models#text-generation-inference

**Comment by vishwanath-gowda** [comment]
I had a similar error, I checked the version of optimum package installed on my system. 
```
$ pip list | grep optimum
optimum                       1.13.2
optimum-neuron                0.0.13
```
I noticed that though I had the latest version of optimum, optimum-neuron was pointing to older github release tag. when I compared the code between the release tags, I noticed the code deviations. 

So I installed the latest version of optimum-neuron which fixed my issue.

```
pip index versions optimum-neuron
pip install optimum-neuron==0.0.13
```

FWIW,

**Comment by maekawataiki** [comment]
GenerationMode is added in this [commit](https://github.com/huggingface/transformers/commit/bff4313b37960751d88ddba0a9fdb9ca5d32a64e) which is available from `transformers==4.32.0` so running `pip install "transformers>=4.32.0"` was required.
